### DISCUSSION TITLE
Getting error "Target revision 79b0bdfc159b3bd1528796a238f9eb68c568297e in Git is not signed, but a signature is required"

### DISCUSSION THREAD

**Discussion by jzaayman** [argoproj/argo-cd#7363] [Q&A]
I'm using ArgoCD v2.1.2+7af9dfb. I've added my GPG signature to ArgoCD and GitHub. My commit is signed and verified (see https://github.com/jjohnson-ah/jjohnson-argocd-tags/commits/main). But when I deploy I get the error "Target revision 79b0bdfc159b3bd1528796a238f9eb68c568297e in Git is not signed, but a signature is required". I can't find any reference to this error in Google. I traced the source code to https://github.com/argoproj/argo-cd/blob/master/controller/state.go#L305, which means `manifestInfo.VerifyResult` is empty, but I have no idea what that means. Does anyone know what I might be doing wrong?

**Comment by jannfis** [comment]
I assume that you have specified a `targetRevision` for your application that is not signed. The error suggests revision `79b0bdfc159b3bd1528796a238f9eb68c568297e`, and according to your repository, this revision is indeed not signed:

![image](https://user-images.githubusercontent.com/3942683/137918486-2cd6af1b-382b-4640-bfa1-4edcc9b363e6.png)

You may want to pin your application to a verified revision instead.
